### ISSUE TITLE
NPE in WebRTCModule/GetUserMediaImpl.java:getUserVideo

### ISSUE THREAD

**Issue by caseygbigelow** [oney/react-native-webrtc#413]
Identical issue to [one raised in another project](https://github.com/atyenoria/react-native-webrtc-janus-gateway/issues/3), looks like it hasn't been raised here so I thought I'd do so.

Using this plugin (great work by the way!) with more recent versions of React Native throws an NPE in `GetUserMediaImpl:getUserVideo` offending block seems to be [lines 393-405](https://github.com/oney/react-native-webrtc/blob/master/android/src/main/java/com/oney/WebRTCModule/GetUserMediaImpl.java#L393).

Fix from the issue linked above:

Change 

```
        // Fall back to defaults if keys are missing.
        int width
                = videoConstraintsMandatory.hasKey("minWidth")
                ? videoConstraintsMandatory.getInt("minWidth")
                : DEFAULT_WIDTH;
        int height
                = videoConstraintsMandatory.hasKey("minHeight")
                ? videoConstraintsMandatory.getInt("minHeight")
                : DEFAULT_HEIGHT;
        int fps
                = videoConstraintsMandatory.hasKey("minFrameRate")
                ? videoConstraintsMandatory.getInt("minFrameRate")
                : DEFAULT_FPS;
```

to 

```
    int width, height, fps;
    // Fall back to defaults if keys are missing.
    if (videoConstraintsMandatory != null) {
         width
                = videoConstraintsMandatory.hasKey("minWidth")
                ? videoConstraintsMandatory.getInt("minWidth")
                : DEFAULT_WIDTH;
        height
                = videoConstraintsMandatory.hasKey("minHeight")
                ? videoConstraintsMandatory.getInt("minHeight")
                : DEFAULT_HEIGHT;
        fps
                = videoConstraintsMandatory.hasKey("minFrameRate")
                ? videoConstraintsMandatory.getInt("minFrameRate")
                : DEFAULT_FPS;
    } else {
        width = DEFAULT_WIDTH;
        height = DEFAULT_HEIGHT;
        fps = DEFAULT_FPS;
    }
```

#### Steps to reproduce the problem

As far as I know, all that's necessary to reproduce is to use the plugin in React Native v0.52.x

I was able to use the plugin in another environment with React Native v0.44.x (based on oney's [`RCTWebRTCDemo` project](https://github.com/oney/RCTWebRTCDemo) without having to make the change to `GetUserMediaImpl`. Not sure exactly what RN version is breaking though.

#### Platform information

* **React Native version**: 0.52.2
* **Plugin version**: 1.58.3
* **OS**: Android
* **OS version**: 7.1.1

**Comment by saghul** [comment]
Sounds about right. Are you up for writing a PR?

**Comment by caseygbigelow** [comment]
[PR opened](https://github.com/oney/react-native-webrtc/pull/414), thanks so much!

**Comment by caseygbigelow** [comment]
Just want to note here that I don't observe the issue when I'm using `react-native-webrtc` [M63 fork](https://github.com/sboily/react-native-webrtc). So this issue might be resolved by #384.
